dataplicity/m2m/wsclient.py

```diff
@@ -208,10 +207,14 @@ def unhandled_error(self, error):
         # ws4py calls this with any socket errors
         # Doesn't matter what the socket error is; connection is fubar.
         log.error('error in WSApp: %s', error)
+        # Can't terminate here
+        # Trick the server in to exiting
+        self.server_terminated = True
```

Comment:
Can't call terminate. It doesn't look like that was intended to be a part of the public interface.

There is a `close` method, which attempts to send a websocket close frame. That wont work if the connection it fubar.

The `server_terminated` boolean exists the main loop. Its undocumented, a bit shitty, but it seems the only way to avoid tracebacks.